src/common/src/types/struct_type.rs

```diff
@@ -117,6 +117,16 @@ impl StructType {
             .chain(std::iter::repeat("").take(self.0.field_types.len() - self.0.field_names.len()))
             .zip_eq_debug(self.0.field_types.iter())
     }
+
+    /// Compares the datatype with another, ignoring nested field names and metadata.
+    pub fn equals_datatype(&self, other: &StructType) -> bool {
+        if self.0.field_types.len() != other.0.field_types.len() {
```

Comment:
nit: zip_eq_fast already contains length check?

Reply:
But it will panic when the length doesn't match, right? We can't panic here.

Reply:
You are right